drivers/display/display_gc9a01a.c

```diff
@@ -0,0 +1,599 @@
+	k_msleep(100);
+	gpio_pin_set_dt(&config->reset, 0);
+	k_msleep(100);
+	gpio_pin_set_dt(&config->reset, 1);
+	k_msleep(100);
+	return 0;
+}
+
+/**
+ * @brief To recover from the sleep mode
+ *
+ * @param dev GC9A01A device instance.
+ * @return 0 when successful, errno otherwise.
+ */
+static int gc9a01a_display_blanking_off(const struct device *dev)
+{
+	LOG_DBG("Turning display blanking off");
+	return gc9a01a_transmit(dev, GC9A01A_DISPON, NULL, 0);
+}
+
+/**
+ * @brief To enter into DISPLAY OFF mode
+ *
+ * @param dev GC9A01A device instance.
+ * @return 0 when successful, errno otherwise.
+ */
+static int gc9a01a_display_blanking_on(const struct device *dev)
+{
+	LOG_DBG("Turning display blanking on");
+	return gc9a01a_transmit(dev, GC9A01A_DISPOFF, NULL, 0);
+}
+
+/**
+ * @brief To set the pixel format
+ *
+ * @param dev GC9A01A device instance.
+ * @param pixel_format Display pixel format
+ * @return 0 when successful, errno otherwise.
+ */
+static int gc9a01a_set_pixel_format(const struct device *dev,
+				    const enum display_pixel_format pixel_format)
+{
+	struct gc9a01a_data *data = dev->data;
+
+	int r;
+	uint8_t tx_data;
+	uint8_t bytes_per_pixel;
+
+	if (pixel_format == PIXEL_FORMAT_RGB_565) {
+		bytes_per_pixel = 2U;
+		tx_data = GC9A01A_PIXSET_MCU_16_BIT | GC9A01A_PIXSET_RGB_16_BIT;
+	} else if (pixel_format == PIXEL_FORMAT_RGB_888) {
+		bytes_per_pixel = 3U;
+		tx_data = GC9A01A_PIXSET_MCU_18_BIT | GC9A01A_PIXSET_RGB_18_BIT;
+	} else {
+		LOG_ERR("Unsupported pixel format");
+		return -ENOTSUP;
+	}
+
+	r = gc9a01a_transmit(dev, GC9A01A_PIXFMT, &tx_data, 1U);
+	if (r < 0) {
+		return r;
+	}
+
+	data->pixel_format = pixel_format;
+	data->bytes_per_pixel = bytes_per_pixel;
+
+	return 0;
+}
+
+/**
+ * @brief To set the display orientation.
+ *
+ * @param dev GC9A01A device instance.
+ * @param orientation Display orientation
+ * @return 0 when successful, errno otherwise.
+ */
+static int gc9a01a_set_orientation(const struct device *dev,
+				   const enum display_orientation orientation)
+{
+	struct gc9a01a_data *data = dev->data;
+
+	int r;
+	uint8_t tx_data = GC9A01A_MADCTL_BGR;
+
+	if (orientation == DISPLAY_ORIENTATION_NORMAL) { // works 0
+		tx_data |= 0;
+	} else if (orientation == DISPLAY_ORIENTATION_ROTATED_90) { // works CW 90
+		tx_data |= GC9A01A_MADCTL_MV | GC9A01A_MADCTL_MY;
+	} else if (orientation == DISPLAY_ORIENTATION_ROTATED_180) { // works CW 180
+		tx_data |= GC9A01A_MADCTL_MY | GC9A01A_MADCTL_MX | GC9A01A_MADCTL_MH;
+	} else if (orientation == DISPLAY_ORIENTATION_ROTATED_270) { // works CW 270
+		tx_data |= GC9A01A_MADCTL_MV | GC9A01A_MADCTL_MX;
+	}
+
+	r = gc9a01a_transmit(dev, GC9A01A_MADCTL, &tx_data, 1U);
+	if (r < 0) {
+		return r;
+	}
+
+	data->orientation = orientation;
+
+	return 0;
+}
+
+/**
+ * @brief To do overall display device configuration.
+ *
+ * @param dev GC9A01A device instance.
+ * @return 0 when successful, errno otherwise.
+ */
+static int gc9a01a_configure(const struct device *dev)
```

Comment:
Maybe call it `_init` instead?

Reply:
I followed the same pattern as the ilitek drivers. There is already a function `gc9a01a_init()` and the` _configure` is called within the `_init`